### DISCUSSION TITLE
How to handle oneshots?

### DISCUSSION THREAD

**Discussion by Inpacchi** [Inpacchi/Manga-Tagger#23] [Ideas]
So - Manga Tagger currently only handles chapters that specifically have some form of chapter, ch, or act in their name. Meaning, if it encounters something without, it just doesn't handle it. I actually wrote a message to have you log an issue when this happens so that we can see if something needs to be changed within Manga Tagger to cover a wider array of chapters.

This has come up in https://github.com/Inpacchi/Manga-Tagger/issues/19 where Manga Tagger cannot properly process oneshot manga chapters. So, I'm polling the community to see how we should go about handling this.

The main concern is that some manga series may have more than one oneshot chapter. In the case of that, how should we go about naming the chapter? An idea I had was to assign each oneshot a number (i.e Horimiya Oneshot 001, Horimiya Oneshot 002, etc.) However this might pull oneshots out of order, if that matters. Also, some may or may not be pulled with titles, so there wouldn't be a good way to identify them.

I went looking at a couple series to get some examples and ideas, but somehow I'm drawing a blank on series that do have oneshots, so can someone please provide some series for me to look at as well?

Looking forward to hearing what all of you have to think!

**Comment by Ablex222** [comment]
It seems that depending on who you ask, oneshot can also mean the same thing as volume. In this case, probably the best and safest solution would be to have it search for the overarching series and have it create the comicinfo.xml, but not let it rename the files. So the process could look something like this. 

1. Detect File xxxxx (oneshot).cbz
2. If "oneshot" present, send send to process B.
3. Process B conducts the search for the title without "oneshot". 
4. Designate oneshots original file title in database and number in database only as encountered. 
5. Insert .xml into file without renaming file. 
6. End results xxxxx (oneshot).cbz

**Reply by Inpacchi** [reply]
I like this solution - although I would prefer a uniform solution when renaming single chapters, it eliminates the need to account for all oneshots. I'll implement this and then we'll see how oneshots work out.
